bundles/statistics/statsgrid2016/FlyoutManager.js

```diff
@@ -0,0 +1,90 @@
+Oskari.clazz.define('Oskari.statistics.statsgrid.FlyoutManager', function (instance) {
+    this.instance = instance;
+    this.sb = instance.sb;
+    this.flyouts = null;
+    this.views = null;
+    this.openFlyouts = [];
+}, {
+    init: function () {
+        var me = this;
+        me.initFlyouts();
+        me.initViews();
+    },
+    open: function( type ) {
+        var me = this;
+        if ( me.openFlyouts[type] ) {
+            me.hide(type);
+            return;
+        }
+
+        var view = me.views[type];
+        var flyout = me.flyouts[type];
+
+        if( view.getElement() === null ) {
+            view.createUi();
+            flyout.makeDraggable({
+                handle : '.oskari-flyouttoolbar, .statsgrid-data-container > .header',
+                scroll : false
+            });
+            flyout.setContent( view.getElement() );
+            var tile = me.instance.plugins['Oskari.userinterface.Tile'];
+            flyout.getElement().find('.icon-close').bind('click', function() {
+                tile.toggleExtensionClass(type, true);
+            });
+        }
+        flyout.move(flyout.options.pos.x, flyout.options.pos.y, true);
+        flyout.show();
+
+        this.openFlyouts[type] = true;
+    },
+    hide: function ( type ) {
+        var me = this;
+        me.flyouts[type].hide();
+        delete me.openFlyouts[type];
+    },
+    initFlyouts: function () {
+        var me = this;
+        me.flyouts = {
+            search: Oskari.clazz.create('Oskari.userinterface.extension.ExtraFlyout', this.instance.getLocalization().tile.title, {
+                width: 'auto',
+                cls: 'statsgrid-search-flyout',
+                view:'search',
+                pos: {
+                    x: 210,
```

Comment:
These should be calculated based on the size of the left bar or you can use the position of the map-container. The following flyouts should get their position from the previous x + width and detect if they are still on screen. The flyouts can overlap if they don't fit the screen side by side.